src/binxml/value_variant.rs

```diff
@@ -207,170 +211,159 @@ impl<'a> BinXmlValue<'a> {
         value_type: &BinXmlValueType,
         cursor: &mut Cursor<&'a [u8]>,
         chunk: Option<&'a EvtxChunk<'a>>,
+        size: Option<u16>,
+        ansi_codec: EncodingRef,
     ) -> Result<BinXmlValue<'a>> {
-        let value = match value_type {
-            BinXmlValueType::NullType => BinXmlValue::NullType,
-            BinXmlValueType::StringType => BinXmlValue::StringType(try_read!(cursor, utf_16_str)),
-            // TODO: these strings need global code-page configuration to work.
-            BinXmlValueType::AnsiStringType => err::UnimplementedToken {
+        trace!("deserialize_value_type: {:?}, {:?}", value_type, size);
+
+        let value = match (value_type, size) {
+            (BinXmlValueType::NullType, _) => BinXmlValue::NullType,
+            (BinXmlValueType::StringType, Some(sz)) => BinXmlValue::StringType(Cow::Owned(
+                read_utf16_by_size(cursor, u64::from(sz))
+                    .context(err::FailedToDecodeUTF16String {
+                        offset: cursor.position(),
+                    })?
+                    .unwrap_or_else(|| "".to_owned()),
```

Comment:
Shouldn't this be an error and not a default fallback?

Reply:
Notice that fallback is on the `Option<String>` part of the `Result<Option<String>>`
Strings are allowed to be empty, but we don't accept I/O errors.